Mcp does not parse Icons correctly if `sizes` is `null`

The following stack trace is produced if the clientInfo received from the client includes an icon without any `sizes` information.

[`sizes` is optional](https://modelcontextprotocol.org/specification/2025-11-25/schema#icon)

Currently we get this error:

```
Stack Dump = jakarta.json.bind.JsonbException: Unable to deserialize property 'clientInfo' because of: Unable to deserialize property 'icons' because of: Problem adapting object of type interface org.mcpjava.server.Icon to class io.openliberty.mcp.internal.requests.IconImpl$IconTO in class class io.openliberty.mcp.internal.requests.IconImpl$Adapter
        at org.eclipse.yasson.internal.deserializer.ObjectDeserializer.deserialize(ObjectDeserializer.java:80)
        at org.eclipse.yasson.internal.deserializer.ObjectDeserializer.deserialize(ObjectDeserializer.java:31)
        at org.eclipse.yasson.internal.deserializer.DefaultObjectInstanceCreator.deserialize(DefaultObjectInstanceCreator.java:57)
        at org.eclipse.yasson.internal.deserializer.DefaultObjectInstanceCreator.deserialize(DefaultObjectInstanceCreator.java:29)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:85)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:34)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:46)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:26)
        at org.eclipse.yasson.internal.DeserializationContextImpl.deserializeItem(DeserializationContextImpl.java:138)
        at org.eclipse.yasson.internal.DeserializationContextImpl.deserialize(DeserializationContextImpl.java:127)
        at org.eclipse.yasson.internal.JsonBinding.deserialize(JsonBinding.java:55)
        at org.eclipse.yasson.internal.JsonBinding.fromJson(JsonBinding.java:62)
        at io.openliberty.mcp.internal.requests.McpRequest.getParams(McpRequest.java:72)
        at io.openliberty.mcp.internal.McpTransport.getParams(McpTransport.java:221)
        at io.openliberty.mcp.internal.McpServlet.initialize(McpServlet.java:570)
        at io.openliberty.mcp.internal.McpServlet.callRequest(McpServlet.java:202)
        at io.openliberty.mcp.internal.McpServlet.doPost(McpServlet.java:146)
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:547)
        at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:614)
        at com.ibm.ws.webcontainer.servlet.ServletWrapper.service(ServletWrapper.java:1262)
        at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:751)
        at com.ibm.ws.webcontainer.servlet.ServletWrapper.handleRequest(ServletWrapper.java:448)
        at com.ibm.ws.webcontainer.filter.WebAppFilterChain.invokeTarget(WebAppFilterChain.java:197)
        at com.ibm.ws.webcontainer.filter.WebAppFilterChain.doFilter(WebAppFilterChain.java:81)
        at com.ibm.ws.webcontainer.filter.WebAppFilterManager.doFilter(WebAppFilterManager.java:1069)
        at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1260)
        at com.ibm.ws.webcontainer.filter.WebAppFilterManager.invokeFilters(WebAppFilterManager.java:1078)
        at com.ibm.ws.webcontainer.servlet.CacheServletWrapper.handleRequest(CacheServletWrapper.java:77)
        at com.ibm.ws.webcontainer40.servlet.CacheServletWrapper40.handleRequest(CacheServletWrapper40.java:87)
        at com.ibm.ws.webcontainer.WebContainer.handleRequest(WebContainer.java:978)
        at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run(DynamicVirtualHost.java:293)
        at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run(HttpDispatcherLink.java:1634)
        at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute(HttpDispatcherLink.java:781)
        at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready(HttpDispatcherLink.java:741)
        at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination(HttpInboundLink.java:574)
        at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest(HttpInboundLink.java:505)
        at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest(HttpInboundLink.java:365)
        at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready(HttpInboundLink.java:332)
        at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators(NewConnectionInitialReadCallback.java:169)
        at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete(NewConnectionInitialReadCallback.java:77)
        at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete(WorkQueueManager.java:513)
        at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO(WorkQueueManager.java:583)
        at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun(WorkQueueManager.java:967)
        at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run(WorkQueueManager.java:1056)
        at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run(ExecutorServiceImpl.java:344)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1090)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:614)
        at java.base/java.lang.Thread.run(Thread.java:1485)
Caused by: jakarta.json.bind.JsonbException: Unable to deserialize property 'icons' because of: Problem adapting object of type interface org.mcpjava.server.Icon to class io.openliberty.mcp.internal.requests.IconImpl$IconTO in class class io.openliberty.mcp.internal.requests.IconImpl$Adapter
        at org.eclipse.yasson.internal.deserializer.ObjectDeserializer.deserialize(ObjectDeserializer.java:80)
        at org.eclipse.yasson.internal.deserializer.ObjectDeserializer.deserialize(ObjectDeserializer.java:31)
        at org.eclipse.yasson.internal.deserializer.DefaultObjectInstanceCreator.deserialize(DefaultObjectInstanceCreator.java:57)
        at org.eclipse.yasson.internal.deserializer.DefaultObjectInstanceCreator.deserialize(DefaultObjectInstanceCreator.java:29)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:85)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:34)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:46)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:26)
        at org.eclipse.yasson.internal.deserializer.DeserializationModelCreator.lambda$typeProcessor$16(DeserializationModelCreator.java:470)
        at org.eclipse.yasson.internal.deserializer.ObjectDeserializer.deserialize(ObjectDeserializer.java:78)
        ... 47 more
Caused by: jakarta.json.bind.JsonbException: Problem adapting object of type interface org.mcpjava.server.Icon to class io.openliberty.mcp.internal.requests.IconImpl$IconTO in class class io.openliberty.mcp.internal.requests.IconImpl$Adapter
        at org.eclipse.yasson.internal.deserializer.AdapterDeserializer.deserialize(AdapterDeserializer.java:45)
        at org.eclipse.yasson.internal.deserializer.DeserializationModelCreator.lambda$typeProcessor$16(DeserializationModelCreator.java:471)
        at org.eclipse.yasson.internal.deserializer.CollectionDeserializer.deserialize(CollectionDeserializer.java:49)
        at org.eclipse.yasson.internal.deserializer.CollectionInstanceCreator.deserialize(CollectionInstanceCreator.java:61)
        at org.eclipse.yasson.internal.deserializer.CollectionInstanceCreator.deserialize(CollectionInstanceCreator.java:37)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:85)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:34)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:46)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:26)
        at org.eclipse.yasson.internal.deserializer.ContextSwitcher.deserialize(ContextSwitcher.java:36)
        at org.eclipse.yasson.internal.deserializer.ContextSwitcher.deserialize(ContextSwitcher.java:22)
        at org.eclipse.yasson.internal.deserializer.ObjectDeserializer.deserialize(ObjectDeserializer.java:78)
        ... 56 more
Caused by: java.lang.NullPointerException: sizes must not be null
        at java.base/java.util.Objects.requireNonNull(Objects.java:246)
        at io.openliberty.mcp.internal.requests.IconImpl.<init>(IconImpl.java:30)
        at io.openliberty.mcp.internal.requests.IconImpl$Adapter.adaptFromJson(IconImpl.java:119)
        at io.openliberty.mcp.internal.requests.IconImpl$Adapter.adaptFromJson(IconImpl.java:112)
        at org.eclipse.yasson.internal.deserializer.AdapterDeserializer.deserialize(AdapterDeserializer.java:43)
        ... 67 more
```

Discovered during integration testing with IBM Bob.

--- a/dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/IconImpl.java
+++ b/dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/IconImpl.java
@@ -12,6 +12,7 @@ package io.openliberty.mcp.internal.requests;
 import static io.openliberty.mcp.internal.EnumAdapters.THEME_ADAPTER;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
@@ -32,12 +33,12 @@ public record IconImpl(String src, String mimeTypeValue, List<String> sizes, The
 
     @Override
     public Optional<String> mimeType() {
-        return Optional.of(mimeTypeValue);
+        return Optional.ofNullable(mimeTypeValue);
     }
 
     @Override
     public Optional<Theme> theme() {
-        return Optional.of(themeValue);
+        return Optional.ofNullable(themeValue);
     }
 
     /**
@@ -113,9 +114,12 @@ public record IconImpl(String src, String mimeTypeValue, List<String> sizes, The
 
         @Override
         public Icon adaptFromJson(IconTO to) throws Exception {
+            if (to.src == null) {
+                return null;
+            }
             return new IconImpl(to.src,
                                 to.mimeType,
-                                to.sizes,
+                                to.sizes == null ? Collections.emptyList() : to.sizes,
                                 to.theme == null ? null : THEME_ADAPTER.adaptFromJson(to.theme));
         }
 

--- a/dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/ImplementationInfoImpl.java
+++ b/dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/requests/ImplementationInfoImpl.java
@@ -82,6 +82,12 @@ public record ImplementationInfoImpl(List<Icon> icons,
 
         @Override
         public ImplementationInfo adaptFromJson(ImplementationInfoTO to) throws Exception {
+            if (to.icons != null && to.icons.contains(null)) {
+                // Remove any unparsable icons
+                to.icons = to.icons.stream()
+                                   .filter(Objects::nonNull)
+                                   .toList();
+            }
             return new ImplementationInfoImpl(to.icons == null ? List.of() : to.icons,
                                               to.name,
                                               to.title,

--- a/dev/io.openliberty.mcp.internal/test/io/openliberty/mcp/internal/test/requests/IconImplTest.java
+++ b/dev/io.openliberty.mcp.internal/test/io/openliberty/mcp/internal/test/requests/IconImplTest.java
@@ -10,8 +10,10 @@
 package io.openliberty.mcp.internal.test.requests;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
 
 import java.util.List;
+import java.util.Optional;
 
 import org.json.JSONException;
 import org.json.JSONObject;
@@ -79,4 +81,21 @@ public class IconImplTest {
         assertEquals(Theme.LIGHT, icon.theme().get());
 
     }
+
+    @Test
+    public void testIconDeserializationMinimal() throws JsonbException, JSONException {
+        JSONObject iconJson = new JSONObject().put("src", TEST_ICON_URL);
+
+        Icon icon = jsonb.fromJson(iconJson.toString(), Icon.class);
+
+        assertEquals(TEST_ICON_URL, icon.src());
+        assertEquals(Optional.empty(), icon.mimeType());
+        assertEquals(List.of(), icon.sizes());
+        assertEquals(Optional.empty(), icon.theme());
+
+        JSONObject emptyIconJson = new JSONObject();
+        Icon emptyIcon = jsonb.fromJson(emptyIconJson.toString(), Icon.class);
+        assertNull(emptyIcon);
+    }
+
 }

--- a/dev/io.openliberty.mcp.internal/test/io/openliberty/mcp/internal/test/requests/ImplementationInfoImplTest.java
+++ b/dev/io.openliberty.mcp.internal/test/io/openliberty/mcp/internal/test/requests/ImplementationInfoImplTest.java
@@ -11,7 +11,9 @@ package io.openliberty.mcp.internal.test.requests;
 
 import static org.junit.Assert.assertEquals;
 
+import java.util.Collections;
 import java.util.List;
+import java.util.Optional;
 
 import org.json.JSONException;
 import org.json.JSONObject;
@@ -84,6 +86,37 @@ public class ImplementationInfoImplTest {
         assertEquals("An example client", info.description().get());
         assertEquals("http://example.com", info.websiteUrl().get());
         assertEquals(List.of(createIcon(32, 32, Theme.LIGHT), createIcon(32, 32, Theme.DARK)), info.icons());
+    }
+
+    @Test
+    public void testImplementationInfoDeserializationMinimal() throws JSONException {
+        JSONObject infoJson = new JSONObject().put("name", "example-client")
+                                              .put("version", "1.0");
+
+        ImplementationInfo info = jsonb.fromJson(infoJson.toString(), ImplementationInfoImpl.class);
+
+        assertEquals("example-client", info.name());
+        assertEquals("example-client", info.title());
+        assertEquals("1.0", info.version());
+        assertEquals(Optional.empty(), info.description());
+        assertEquals(Optional.empty(), info.websiteUrl());
+        assertEquals(Collections.emptyList(), info.icons());
+    }
+
+    @Test
+    public void testImplementationInfoDeserializationInvalidIcons() throws JSONException {
+        JSONObject infoJson = new JSONObject().put("name", "example-client")
+                                              .put("version", "1.0")
+                                              .put("icons", List.of(new JSONObject().put("wibble", "foo")));
+
+        ImplementationInfo info = jsonb.fromJson(infoJson.toString(), ImplementationInfoImpl.class);
+
+        assertEquals("example-client", info.name());
+        assertEquals("example-client", info.title());
+        assertEquals("1.0", info.version());
+        assertEquals(Optional.empty(), info.description());
+        assertEquals(Optional.empty(), info.websiteUrl());
+        assertEquals(Collections.emptyList(), info.icons());
     }
 
     public Icon createIcon(int width, int height, Theme theme) {

